doc/feature/restAPI.md

```diff
@@ -0,0 +1,94 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented,
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+Invoking a GET request on the / url returns the list of top-level namespaces, 
+e.g. 'zerobuf'. Similarily, a GET /zerobuf/render/ would return the objects in
+the zerobuf::render namespace. Finally, a GET /namespace/object/schema would 
+return the schema for the object itself.
+
+   Request: GET http://hostname:port/
+   Response: [ "namespace1", "namespace2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/
+   Response: [ "object1", "object2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/object1/schema
+   Response: { "object1" : { "attribute1": "string", "attribute2": "boolean" } }
```

Comment:
I've added some indications in the Implementation section of the document. I am voting for the 'by default' option. This mechanism should be transparent for the user of the class.